pkg/client/restclient/config.go

```diff
@@ -50,6 +50,8 @@ type Config struct {
 	// be appended to all request URIs used to access the apiserver. This allows a frontend
 	// proxy to easily relocate all of the apiserver endpoints.
 	Host string
+	// UnixSocket is a path to the unix socket that the (proxy) apiserver is listening on.
+	UnixSocket string
```

Comment:
A brief search shows at least one instance of setting the scheme to "http+unix": https://github.com/msabramo/requests-unixsocket

Changing the scheme (not the host) is the thing I would have expected.